File: driver/examples/request.pl
use IO::S4::Client;
use Data::Dumper;

$sn = $ARGV[0];
$cn = $ARGV[1];

$c = new IO::S4::Client("localhost", 2334);
$c->init();
print Dumper($c);

$c->connect({'readMode' => 'private'});

while (<STDIN>) {
  chomp;
  print "Sending request\n";
  $c->send($sn, $cn, $_);

  select(undef, undef, undef, 0.1);
  print "Waiting for response...\n";


  my $l = $c->recv();
  print "Response: " . Dumper($l);
}

$c->disconnect();

File: s4-driver/examples/src/main/resources/scripts/request.py
import io.s4.client.driver
import pprint;
import sys;

mode = {'readMode': 'private', 'writeMode': 'enabled'};

stream = sys.argv[1];
clazz = sys.argv[2];

d = io.s4.client.driver.Driver('localhost', 2334)
d.setDebug(True);

d.initialize();
d.connect(mode);

print "Sending all requests..."

for req in sys.stdin.readlines():
    d.send(stream, clazz, req);

print "Waiting 5 sec to collect all responses..."
responses = d.recvAll(5);
print "\n"*4

print "Done. Results:"
print pprint.pformat(responses, indent=4);

d.disconnect();
